src/main/java/org/pentaho/di/trans/dataservice/www/RepositoryPublishServlet.java

```diff
@@ -0,0 +1,58 @@
+/*! ******************************************************************************
+ *
+ * Pentaho Data Integration
+ *
+ * Copyright (C) 2002-2016 by Pentaho : http://www.pentaho.com
+ *
+ *******************************************************************************
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ ******************************************************************************/
+
+package org.pentaho.di.trans.dataservice.www;
+
+import org.osgi.framework.BundleContext;
+import org.pentaho.di.core.annotations.CarteServlet;
+import org.pentaho.di.www.BaseCartePlugin;
+import org.pentaho.osgi.kettle.repository.locator.api.KettleRepositoryProvider;
+
+import java.io.IOException;
+
+/**
+ * Created by bmorrise on 9/9/16.
+ */
+@CarteServlet(
+  id = "repoPublish",
+  name = "Repository Publish",
+  description = "Publish the accessible servlet repository" )
+public class RepositoryPublishServlet extends BaseCartePlugin {
+
```

Comment:
We need a way to provide a repository provider to the server, so after talking with Nick, we had this idea of creating a servlet whose responsibility it is to publish the service that would provide the repository the same way we'd done it before. We tried other solutions, but this seemed to be the most straight forward. I talked to a few people about the best way to get the repository and it seemed like pulling it from the transformationMap was the best way to go. I'm definitely open to better alternatives.